pyqg/xarray_output.py

```diff
@@ -0,0 +1,152 @@
+        'y': {'long_name': 'real space grid points in the y direction', 'units': 'grid point',},
+        'l': {'long_name': 'spectal space grid points in the l direction', 'units': 'meridional wavenumber',},
+        'k': {'long_name': 'spectal space grid points in the k direction', 'units': 'zonal wavenumber',},
+        'nx': {'long_name': 'number of real space grid points in x direction', 'units': 'none',},
+        'ny': {'long_name': 'number of real space grid points in y direction (default: nx)', 'units': 'none',},
+        'nz': {'long_name': 'number of vertical levels', 'units': 'none',},
+        'nl': {'long_name': 'number of spectral space grid points in l direction', 'units': 'grid point',},
+        'nk': {'long_name': 'number of spectral space grid points in k direction', 'units': 'grid point',},
+        'rek': {'long_name': 'linear drag in lower layer', 'units': 'seconds\u207B\u00B9',},
+        'tc': {'long_name': 'model timestep', 'units': 'seconds',},
+        'dt': {'long_name': 'numerical timestep', 'units': 'seconds',},
+        'L': {'long_name': 'domain length in x direction', 'units': 'meters',},
+        'W': {'long_name': 'domain length in y direction', 'units': 'meters',},
+        'filterfac': {'long_name': 'amplitude of spectral spherical filter', 'units': '',},
+        'twrite': {'long_name': 'interval for cfl writeout', 'units': 'number of timesteps',},
+        'tmax': {'long_name': 'total time of integration', 'units': 'seconds',},
+        'tavestart': {'long_name': 'start time for averaging', 'units': 'seconds',},
+        'tsnapstart': {'long_name': 'start time for snapshot writeout', 'units': 'seconds'},
+        'taveint': {'long_name': 'time interval for accumulation of diagnostic averages', 'units': 'seconds'},
+        'tsnapint': {'long_name': 'time interval for snapshots', 'units': 'seconds',},
+        'ntd': {'long_name': 'number of threads used', 'units': 'none',},
+        'pmodes': {'long_name': 'vertical pressure modes', 'units': 'none',},
+        'radii': {'long_name': 'deformation radii', 'units': 'meters',},
+    }
+
+    # dict for dataset attributes
+    ds_attr_database = {
+        'title': 'pyqg: Python Quasigeostrophic Model',
+        'institution': '',
+        'source': ('version: {}'.format(pyqg.__version__)),
+        'history': '',
+        'references': 'https://pyqg.readthedocs.io/en/latest/index.html',
+        'comment': '', 
+    }
+
+
+    # Create list of variables
+    variables = {}
+    for vname in dim_database:
+        data = getattr(m,vname)
+        if 'time' in dim_database[vname]:
+            variables[vname] = (dim_database[vname],data[np.newaxis,...])
+        else:
+            variables[vname] = (dim_database[vname],data)
+
+    # Create list of coordinates
+    coordinates = {}
+    cname_to_1D = ['time','z','x','y','l','k']
+    coords_1D = [(np.array([m.t])), (np.arange(1,m.nz+1)), (m.x[0,:]), (m.y[:,0]), (m.l[:,0]), (m.k[0,:])] 
```

Comment:
Let's see if we can avoid hard coding these transformations. Aren't the 1D versions of the arrays available somewhere as part of `m`?

Reply:
you could do something like this

```python
transformations = {
    'k': lambda x: x.k[0, :]
}

# more verbose verion - does exactly the same thing
def get_k_from_model(m):
    return m.k[0, :]
transformations = {
    'k': get_k_from_model
}



# then inside the function
for k, func in transformations.items():
    data[k] = func(m)
```